zstash/globus.py

```diff
@@ -49,18 +78,22 @@ def check_endpoint_version_5(ep_id):
     return False
 
 
-def submit_transfer_with_checks(transfer_data):
+def submit_transfer_with_checks(globus_info: GlobusInfo):
+    if not globus_info.transfer_client:
+        raise ValueError("transfer_client is undefined")
     try:
-        task = transfer_client.submit_transfer(transfer_data)
+        task = globus_info.transfer_client.submit_transfer(globus_info.transfer_data)
     except TransferAPIError as err:
         if err.info.consent_required:
             scopes = "urn:globus:auth:scope:transfer.api.globus.org:all["
-            for ep_id in [remote_endpoint, local_endpoint]:
-                if check_endpoint_version_5(ep_id):
+            for ep_id in [globus_info.remote_endpoint, globus_info.local_endpoint]:
+                if check_endpoint_version_5(globus_info, ep_id):
                     scopes += f" *https://auth.globus.org/scopes/{ep_id}/data_access"
             scopes += " ]"
-            native_client = NativeClient(
-                client_id="6c1629cf-446c-49e7-af95-323c6412397f", app_name="Zstash"
+            native_client = NativeClient(client_id=ZSTASH_CLIENT_ID, app_name="Zstash")
+            log_current_endpoints(globus_info)
+            logger.debug(
+                "submit_transfer_with_checks. Calling login, which may print 'Please Paste your Auth Code Below:'"
             )
             native_client.login(requested_scopes=scopes)
             # Quit here and tell user to re-try
```

Comment:
This `"Consents added, please re-run the previous command to start transfer"` is a real obstacle to users. It basically means they have to run a toy version of `zstash` first, as specified in https://github.com/E3SM-Project/zstash/discussions/329 and https://github.com/E3SM-Project/zstash/issues/339. We need to at least try to immediately alert the user that consents aren't set up rather than waiting until we try to transfer data.

Reply:
Agreed.  There should be a globus function specific to this issue..

```
def has_consent(endpoint, path, consent_type):
    . . .
    return [True, False]
```

Reply:
@forsyth2 Without doing "damage", is there a way we could code a module (or function) that emulates "has_consent()" above, going as far as obtaining a transfer client (for a "toy" transfer, like "ls") and hides the details from the user?  

(I don't really understand what is gained by registering for a "client_id".  Is this intended to shorten the authentication steps?)  If I write an application that has no registered client_id, can I still use the globus_sdk to code up functioning?

Reply:
Yeah, I'm trying to do some prototyping of what that would look like.